### PR TITLE
Feat/feature screenone data module

### PR DESCRIPTION
**Pull request by anarahari**

### CODE REVIEW

File: common/src/main/java/com/compose/common/CommonConstants.kt [hunk 1]
```diff
@@ -0,0 +1,5 @@
+package com.compose.common
+
+object CommonConstants {
```
Comment (shreyansh-mm) [inline common/src/main/java/com/compose/common/CommonConstants.kt]: Should be marked as internal

File: data/src/main/java/com/compose/data/repository/CharacterRepositoryImpl.kt [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+import androidx.paging.PagingConfig
+import androidx.paging.PagingData
+import com.apollographql.apollo.ApolloClient
+import com.common.graphql.GetCharactersQuery
+import com.compose.domain.paging.CharacterPagingSource
+import com.compose.domain.repository.CharacterRepository
+import kotlinx.coroutines.flow.Flow
+import javax.inject.Inject
+
+class CharacterRepositoryImpl @Inject constructor(private val apolloClient: ApolloClient) : CharacterRepository {
```
Comment (shreyansh-mm) [inline data/src/main/java/com/compose/data/repository/CharacterRepositoryImpl.kt]: Let's see if we can make this class `internal`

**Review by shreyansh-mm — CHANGES_REQUESTED** [review]

File: data/src/main/java/com/compose/data/repository/CharacterRepositoryImpl.kt (hunk 2 above)
Comment (anarahari) [reply data/src/main/java/com/compose/data/repository/CharacterRepositoryImpl.kt]: ok let me try

File: common/src/main/java/com/compose/common/CommonConstants.kt (hunk 1 above)
Comment (anarahari) [reply common/src/main/java/com/compose/common/CommonConstants.kt]: done

File: data/src/main/java/com/compose/data/datamapper/DataMapper.kt [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+import com.compose.domain.mapper.CharacterModel
+
+fun getCharactersQueryToCharacterModel(results: GetCharactersQuery.Result?): CharacterModel {
+    return CharacterModel(
+        id = results?.character?.id.orEmpty(),
+        name = results?.character?.name.orEmpty(),
+        image = results?.character?.image.orEmpty(),
+        gender = results?.character?.gender.orEmpty(),
+        species = results?.character?.species.orEmpty(),
+        status = results?.character?.status.orEmpty()
```
Comment (shreyansh-mm) [inline data/src/main/java/com/compose/data/datamapper/DataMapper.kt]: why do we have or empty?

File: data/src/main/java/com/compose/data/paging/CharacterPagingSource.kt [hunk 4]
```diff
@@ -17,19 +16,19 @@ class CharacterPagingSource(
...
                 val characters = response.data?.characters?.results?.filterNotNull() ?: emptyList()
-                val nextKey =
-                    if (response.data?.characters?.info?.next != null) {
+                val nextKey = page + 1
+                    /*if (response.data?.characters?.info?.next != null) {
                         page + 1
                     } else {
                         null
-                    }
+                    }*/
```
Comment (shreyansh-mm) [inline data/src/main/java/com/compose/data/paging/CharacterPagingSource.kt]: This is being confusing to read, lets not keep code commented, remove them so it becomes easier to read.

File: data/src/main/java/com/compose/data/repository/CharacterRepositoryImpl.kt [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+import kotlinx.coroutines.flow.flow
+import javax.inject.Inject
+
+internal class CharacterRepositoryImpl @Inject constructor(private val apolloClient: ApolloClient) :
+    CharacterRepository {
+    override suspend fun getCharacters(): Flow<List<CharacterModel>> = flow {
+        val response = apolloClient.query(GetCharactersQuery()).execute()
+        if (response.hasErrors()) {
+            emit(emptyList())
+            return@flow
```
Comment (shreyansh-mm) [inline data/src/main/java/com/compose/data/repository/CharacterRepositoryImpl.kt]: I don't think we need to use this, we can update the conditions above and make our code cleaner

**Review by shreyansh-mm — CHANGES_REQUESTED** [review]

File: data/src/main/java/com/compose/data/datamapper/DataMapper.kt (hunk 3 above)
Comment (anarahari) [reply data/src/main/java/com/compose/data/datamapper/DataMapper.kt]: orEmpty method checks if the string value is null, if the value is null, an empty string is returned

**Pull request merged**